kuzzleio/kuzzle
# Use local file to build kuzzle and plugin-dev images

## What does this PR do ?




The actual way to build images didn't work properly for releases.  
I was using `git clone -b <branchname>` to get Kuzzle code and plugins but this didn't work with tagged release because we didn't have a branch for each of them.

This PR use the code of the repository to build the `kuzzleio/kuzzle` and `kuzzleio/plugin-dev` images.  
In this way the build process is much simpler.  
For debugging and local build purpose, this script support 3 modes set by an environment variable:
  - `MODE=test` (or empty string): Just print the command that will be run in other modes
  - `MODE=local`: Build and tag images locally
  - `MODE=production`: Build and tag images then push them to Dockerhub

Historically we choose to get the code from git so we can have more flexibility about what plugin we wan't inside Kuzzle. We were also capable to build every version of Kuzzle just by running the script and specify a branch name.  
This was cool but this made the workflow much complicated.

:pray: This is the last PR about Node 8 images I hope

### How should this be manually tested?

Test only the workflow for merging on `*-dev` (nothing will be build without `MODE=production`)
  - Step 1 : `TRAVIS_BRANCH=1-dev bash build-docker-images.sh`
  - Step 2 : Should tell you that `kuzzle` and `plugin-dev` images are built, and tagged with `1-dev` and `develop`
  - Step 3 : `TRAVIS_BRANCH=2-dev bash build-docker-images.sh`
  - Step 4 : Should tell you that `kuzzle` and `plugin-dev` images are built, and tagged with `2-dev`

Test only the workflow for release 
  - Step 1 : `RELEASE_TAG=1.42 bash build-docker-images.sh`
  - Step 2 : Should tell you that `kuzzle` and `plugin-dev` images are built, and tagged with `1.42` and `latest`
  - Step 3 : `RELEASE_TAG=2.0-beta bash build-docker-images.sh`
  - Step 4 : Should tell you that `kuzzle` and `plugin-dev` images are built, and tagged with `2.0-beta`

Test real build without pushing images to Dockerhub:
  - Step 1 : `MODE=local RELEASE_TAG=1.42 bash build-docker-images.sh`
  - Step 2 : Check `docker images`
  - Step 3 : `MODE=local TRAVIS_BRANCH=1-dev bash build-docker-images.sh`
  - Step 4 : Check `docker images`

### Other changes

  - add debug mode to script
  - prepare for `2-dev` branch
